### PR TITLE
leases stuck on updating fix

### PR DESCRIPTION
**Pull request by mkuzdowicz**
## What does this change?

Bug fix for leases stuck on updating 

trello card: https://trello.com/c/FIoSr7L1/157-bug-small-leases-stuck-on-updating-and-never-resolves-until-page-refresh

## How can success be measured?
- kahuna have unit tests setup
- after adding lease frontend do not end-up with updating... but it display added leases

## Tested?
- [x] on TEST
- [x] unit tests

### CODE REVIEW

File: kahuna/public/js/services/api/leases-helper.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+function readLeases(image) {
+  return image.data.leases.data;
+}
+
+export function getApiImageAndApiLeasesIfUpdated(image, apiImage) {
+    const apiLeases = readLeases(apiImage);
+    const leases = readLeases(image);
+    const apiImageAndApiLeases = {image: apiImage, leases: apiLeases};
+    const isNewlyCreated = leases.lastModified === null;
+    if (isNewlyCreated) {
```
Comment (jonathonherbert) [inline kahuna/public/js/services/api/leases-helper.js]: No pattern matching/guards in JS :( One day ...

File: kahuna/public/js/services/api/leases-helper.spec.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            }
+          }
+        }
+      };
+
+      const actual = getApiImageAndApiLeasesIfUpdated(clientSideImage, apiImageWithUpToDateLease);
+
+      const expected = {image: apiImageWithUpToDateLease, leases: apiImageWithUpToDateLease.data.leases.data};
+
+      expect(actual).toEqual(expected);
```
Comment (jonathonherbert) [inline kahuna/public/js/services/api/leases-helper.spec.js]: Good to see these tests, thanks for setting them up, it's very good for the project 👍

File: kahuna/public/js/services/api/leases.js [hunk 3]
```diff
@@ -112,13 +112,13 @@ leaseService.factory('leaseService', [
       const imagesArray = images.toArray ? images.toArray() : images;
       return $q.all(imagesArray.map(image => {
         return image.get().then(apiImage => {
-          const apiLeases = imageAccessor.readLeases(apiImage);
-          const leases = imageAccessor.readLeases(image);
-          const currentLastModified = moment(apiLeases.lastModified);
-          const previousLastModified = moment(leases.lastModified);
-          if (currentLastModified.isAfter(previousLastModified)) {
-            return { image: apiImage, leases: apiLeases };
+          const apiImageAndApiLeases = getApiImageAndApiLeasesIfUpdated(image, apiImage);
```
Comment (jonathonherbert) [inline kahuna/public/js/services/api/leases.js]: This is now much clearer behind a function call, excellent 👍

Comment (mkuzdowicz) [reply kahuna/public/js/services/api/leases.js]: thanks :)

File: kahuna/public/js/services/api/leases-helper.spec.js (hunk 2 above)
Comment (mkuzdowicz) [reply kahuna/public/js/services/api/leases-helper.spec.js]: i am happy about that as well :)

File: kahuna/public/js/services/api/leases-helper.js (hunk 1 above)
Comment (mkuzdowicz) [reply kahuna/public/js/services/api/leases-helper.js]: would be usefull to move that to TypeScript one day

**Pull request merged**